backend/src/main/java/com/pokemon/pokedex/graphql/GraphQLProvider.java

```diff
@@ -0,0 +1,60 @@
+package com.pokemon.pokedex.graphql;
+
+import static graphql.schema.idl.TypeRuntimeWiring.newTypeWiring;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import graphql.GraphQL;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.RuntimeWiring;
+import graphql.schema.idl.SchemaGenerator;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import java.io.IOException;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GraphQLProvider {
+
+  @Autowired
+  GraphQLPokemonDataFetchers graphQLPokemonDataFetchers;
```

Comment:
don't prefer constructor injection?

Reply:
Depending on situation. I tend to use constructor injection more in testing but in this specific case, again, running out of time this was how the example for the use of grapqh was and i didn't check it thoroughly